### PR TITLE
Add exception handler in Update Product mutation for collection with empty ID

### PR DESCRIPTION
**Pull request by dexon44**
I want to merge this change because it add exception handler in Product Update mutation for collection with empty ID

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2856 Unhandled exception in product update mutation for collection with empty ID](https://app.clickup.com/t/2549495/SALEOR-2856)

**Review by tomaszszymanski129 — APPROVED** [review]
Two minor things:
- Typo in title (what is `Updpod`?!)
- Misleading title, you haven't add the exception handler, just fixed it's behavior with empty IDs.

**Review by fowczarek — COMMENTED** [review]
Could we add a test for this case?

File: saleor/graphql/product/tests/test_product.py [hunk 1]
```diff
@@ -5546,6 +5546,42 @@ def test_update_product_with_page_reference_attribute_value(
...
+      id
+    }
+  }
+}
+
+    """
+    product_id = graphene.Node.to_global_id("Product", product.pk)
+    variables = {
+        "productId": product_id,
+        "input": {"collections": ""},
```
Comment (maarcingebala) [inline saleor/graphql/product/tests/test_product.py:5571]: Wasn't it `collections: [""]` that was failing (empty string, but inside the list)?

File: saleor/graphql/product/tests/test_product.py [hunk 2]
```diff
@@ -5546,6 +5546,42 @@ def test_update_product_with_page_reference_attribute_value(
...
+    productErrors {
+      field
+      message
+      code
+    }
+    product {
+      id
+    }
+  }
+}
```
Comment (fowczarek) [inline saleor/graphql/product/tests/test_product.py:5565]:
```suggestion
    query = """
    mutation updateProduct($productId: ID!, $input: ProductInput!) {
      productUpdate(id: $productId, input: $input) {
        productErrors {
          field
          message
          code
        }
        product {
          id
        }
      }
    }
```

**Review by eCzerniak — APPROVED** [review]

File: saleor/graphql/product/tests/test_product.py [hunk 3]
```diff
@@ -5546,6 +5546,42 @@ def test_update_product_with_page_reference_attribute_value(
...
+    # when
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_products]
+    )
+    # then
+    content = get_graphql_content(response)
+    data = content["data"]["productUpdate"]
+    assert len(data["productErrors"]) == 1
+    productErrors = data["productErrors"][0]
+    assert productErrors["code"] == "GRAPHQL_ERROR"
```
Comment (IKarbowiak) [inline saleor/graphql/product/tests/test_product.py:5582]: In tests, we are using snake case so for consistency it should be:
```suggestion
    product_errors = data["productErrors"][0]
    assert product_errors["code"] == "GRAPHQL_ERROR"
```

Also, we usually compare error code to value from the error code enum, so in this case it should be:
```suggestion
    assert product_errors["code"] == ProductErrorCode.GRAPHQL_ERROR.name
```

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**